src/main/webapp/app/upload/upload.component.ts

```diff
@@ -1,23 +1,103 @@
-import { Component } from '@angular/core';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
 import { Router } from '@angular/router';
-import { UploadEvent, UploadFile, FileSystemFileEntry } from 'ngx-file-drop';
+import { FileSystemFileEntry, UploadEvent, UploadFile } from 'ngx-file-drop';
 import { UploadService } from './upload.service';
+import { Pool } from 'app/pool/pool';
+import { PoolListService } from 'app/pool/pool-list/pool-list.service';
+
+enum PoolType {
+    New,
+    Old
+}
 
 @Component({
     selector: 'jhi-home',
     templateUrl: './upload.component.html',
-    styleUrls: ['upload.css']
+    styleUrls: ['upload.css'],
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class UploadComponent {
     error: string;
     uploadResponse = { status: '', message: '' };
 
+    currentPool: Map<String, Pool> = new Map<String, Pool>();
+    newPool: Map<String, Pool> = new Map<String, Pool>();
+
+    public poolType = PoolType;
+    public isCollapsed = true;
     public files: UploadFile[] = [];
-    file: File;
 
-    constructor(private uploadService: UploadService, private router: Router) {}
+    public currentConflictPool: Pool[] = [];
+    public newConflictPool: Pool[] = [];
+    public merged: Pool[] = [];
+    public choosePattern = new Map();
+    public file: File;
 
-    onSubmit() {
+    constructor(
+        private uploadService: UploadService,
+        private router: Router,
+        private poolService: PoolListService,
+        private cd: ChangeDetectorRef
+    ) {}
+
+    async onSubmit() {
+        if (!this.file) {
+            return (this.error = 'Please insert file');
+        }
+        this.clearField();
+        const formData = new FormData();
+        formData.append('file', this.file);
+
+        const oldOnesPromise = this.poolService.getPools().toPromise();
+        const newOnesPromise = this.uploadService.parse(formData).toPromise();
+
+        (await oldOnesPromise).body.forEach(pool => this.currentPool.set(pool.poolId, pool));
+        (await newOnesPromise).body.forEach(pool => this.newPool.set(pool.poolId, pool));
+        this.getConflicts();
+        this.isCollapsed = false;
+        this.cd.detectChanges();
+        this.newConflictPool.forEach((pool, _) => this.choosePattern.set(pool.poolId, this.poolType.New));
+    }
+
+    onSave() {
+        this.choosePattern.forEach((value, key) => {
+            let item;
+            if (value === this.poolType.New) {
+                item = this.newPool.get(key);
+            } else {
+                item = this.currentPool.get(key);
+            }
+            this.merged.push(item);
+        });
+
+        const formData = new FormData();
+        formData.append('pools', JSON.stringify(this.merged));
```

Comment:
You don't have to convert it to FormData, it can be sent as plain JSON making things easier on the server side as Spring automatically maps JSON to objects ( I think so).

Reply:
fixed, now JSON is being sent, moreover spring do actually auto parse JSONs to objects.